pinot-query-planner/src/main/java/org/apache/pinot/query/planner/stage/MailboxReceiveNode.java

```diff
@@ -43,11 +57,25 @@ public MailboxReceiveNode(int stageId) {
 
   public MailboxReceiveNode(int stageId, DataSchema dataSchema, int senderStageId,
       RelDistribution.Type exchangeType, @Nullable KeySelector<Object[], Object[]> partitionKeySelector,
+      @Nullable List<RelFieldCollation> fieldCollations, boolean isInputSorted, boolean sortOnReceiver,
```

Comment:
+1 but i dont think this PR would be related since the refactoring only deal with visitor invariant, shared info across the stage such as partition, stageID, etc. 

but i would anticipate a rebase issue. thank you for calling it out @ankitsultana

Reply:
+1 this will only help with future shared info. The fields I've added as part of this PR aren't shared.